pkg/component/controller/konnectivity.go

```diff
@@ -337,6 +341,11 @@ spec:
           imagePullPolicy: {{ .PullPolicy }}
           name: konnectivity-agent
           command: ["/proxy-agent"]
+          env:
+              # the variable is not in a use
+              # we need it to have agent restarted on server count change
+              - name: EPHEMERAL_SERVER_COUNT
```

Comment:
We can also add some prefix (`K0S_`) to indicate that it's used by k0s and not related to the konnectivity options.